SCClassLibrary/Common/Core/Error.sc

```diff
@@ -147,12 +147,28 @@ ShouldNotImplementError : MethodError {
 }
 
 DoesNotUnderstandError : MethodError {
-	var <>selector, <>args;
+	var <>selector, <>args, <suggestedCorrection, suggestion;
 	*new { arg receiver, selector, args;
-		^super.new(nil, receiver).selector_(selector).args_(args)
+		^super.new(nil, receiver).selector_(selector).args_(args).init
+	}
+	init {
+		var methods, methodNames, editDistances, minIndex;
+		methods = receiver.class.superclasses.add(receiver.class).collect(_.methods).reduce('++');
+		methodNames = methods.collect { |x| x.name.asString };
+		editDistances = methodNames.collect(_.editDistance(selector.asString));
+		minIndex = editDistances.minIndex;
+		// Edit distance of 3 chosen arbitrarily; also, filter out completely dissimilar matches
+		// to avoid unhelpful suggestions.
+		if(editDistances[minIndex] <= 3 and: { methodNames[minIndex].similarity(selector.asString) > 0 }) {
+			suggestedCorrection = methods[minIndex];
+			suggestion = " Did you mean '" ++ methodNames[minIndex] ++ "'?";
+		} {
```

Comment:
would it be better to collect the similarities, and use maxIndex, rather than the edit distances?

that way, the if becomes:
```supercollider
// if our strongest match is at least 80% similar
if(similarities[maxIndex] > 0.8) {
    // make suggestion
}
```
and we have a more meaningful comparison, rather than an arbitrary number of edits?
It also makes it a little easier to tweak?

Reply:
as i was testing this i found edit distance to be more useful as a first approximation than similarity. for instance `size` and `szie` have a similarity of just `0.5`.

Reply:
I'd agree with that, from what I've seen I think it would totally reasonable for 0.5 to be the value for the comparison. 
Having said that, the semantics are currently "you can make three mistakes", which is equally clear

Reply:
i think ultimately user feedback is likely to help a lot in guiding improvements here. i think (hope) if we get this out into a release or even just on the develop branch for awhile we'll have people making suggestions.